invenio_drafts_resources/resources/records/errors.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Error classes and handlers."""
+
+from flask import redirect
+
+
+def create_latest_version_redirect_error_handler():
+    """Creates an error handler for `PIDRedirectedError` error."""
+
+    def latest_version_redirected_error_handler(e):
+        response = redirect(e.location, code=301)
+        response.headers['Location'] = e.location
+        return response
+
+    return latest_version_redirected_error_handler
+
+
+class LatestVersionRedirect(Exception):
```

Comment:
Minor: Can this not be an [HTTPJSONException](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/errors.py#L45)



```python 
def api_redirect(location, code=301):
     raise RedirectException(location, code=code)


class RedirectException(HTTPJSONException):
    code = 301

    def __init__(self, location, **kwargs):
        self.location = location
        kwargs.setdefault(description="Redirecting...")
        super().__init__(**kwargs)
    
    def get_headers(self, environ=None):
        return [("Content-Type", "application/json"), ("Location", self.location)]

    def get_body(self, environ=None):
        """Get the request body."""
        body = {"status": self.code, "message": self.get_description(environ), "location": self.location}
        return json.dumps(body)
```

Then you can completely avoid the error map in the service config.